File: medium/0019_Remove_Nth_Node_From_End_Of_List.c
struct ListNode* removeNthFromEnd(struct ListNode* head, int n) {
    if(head == NULL || head->next == NULL){
        return NULL;
    }

    struct ListNode* dummy = malloc(sizeof(struct ListNode));
    dummy->next = head;
    struct ListNode *fast = dummy;
    struct ListNode *slow = dummy;

    while(n>=0){
        fast = fast->next;
        n -= 1;
    }

    while(fast != NULL){
        fast = fast->next;
        slow = slow ->next;
    }

    slow->next = slow->next->next;

    return dummy->next;

}

File: medium/0019_Remove_Nth_Node_From_End_Of_List.py
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        
        if not head or not head.next:
            return None
        
        dummy = ListNode(0, head)
        fast = dummy
        slow = dummy


        for _ in range(n+1):
            fast = fast.next
        
        while fast:

            fast = fast.next
            slow = slow.next
        
        slow.next = slow.next.next
        
        return dummy.next
